packages/components/src/components/SquiggleEditor.tsx

```diff
@@ -164,33 +164,43 @@ export let SquigglePartial: React.FC<SquigglePartialProps> = ({
     xyPointLength: outputXYPoints,
   };
   let [expression, setExpression] = React.useState(initialSquiggleString);
-  let squiggleResult = runPartial(
-    expression,
-    bindings,
-    samplingInputs,
-    jsImports
-  );
-  if (squiggleResult.tag == "Ok") {
-    if (onChange) onChange(squiggleResult.value);
-  }
+  let [error, setError] = React.useState<string | null>(null);
+
+  // Runs squiggle and updates state/calls props apprropriately
+  let runSquiggle = (newExpression: string) => {
```

Comment:
This stuff (runSquiggle vs. onChange vs. setError) feels a bit hacky. At least with old school React, we could use props and organize them, plus separate components into clear items with few dependencies. Not a huge concern to fix with this PR, if it seems pretty hard to, but worth consideration.